packages/components/input/src/input.js

```diff
@@ -26,28 +49,52 @@ export class TSInput extends TSElement {
 		if (this.disabled !== undefined && this.disabled !== null && input) {
 			input.disabled = true;
 		}
+		if (this.readonly !== undefined && this.readonly !== null && input) {
+			input.readonly = true;
+		}
+		if (input.tagName === 'TEXTAREA') {
+			input.addEventListener('input', this.calcTextareaHeight);
+		}
+	}
+
+	disconnectedCallback() {
+		const input = this.getSlotInput();
+		if (input.tagName === 'TEXTAREA') {
+			input.removeEventListener('input', this.calcTextareaHeight);
+		}
+	}
+
+	calcTextareaHeight(event) {
+		const el = event.target;
+		el.style.cssText = 'height:auto; padding: 10px;'; // to let it get scroll, so we can find the value
+		el.style.cssText = `height: ${el.scrollHeight + 2}px`; // 2 is the border width
 	}
 
 	getSlotInput() {
 		const slot = this.shadowRoot.querySelector('slot');
 		if (!slot) {
 			return null;
 		}
-		const elements = slot.assignedNodes().filter(el => el.constructor.name === 'HTMLInputElement');
+		const elements = slot
+			.assignedNodes()
+			.filter(el => ['HTMLInputElement', 'HTMLTextAreaElement'].includes(el.constructor.name));
 		if (elements.length < 1) {
-			console.warn('We need one input element in the slot');
+			console.warn('We need one input or textarea element in the slot');
 		}
 		if (elements.length > 1) {
-			console.warn('We need only one input element in the slot');
+			console.warn('We need only one input or one textarea element in the slot');
 		}
 		return elements[0];
 	}
 
 	render() {
 		return html`
-			<div>
+			<div dir=${this.direction}>
+				${this.iconLeft &&
+				html` <ts-icon class="icon-left" icon=${this.iconLeft} size="large" type="action"></ts-icon> `}
```

Comment:
nit: why there are spaces at start and end in string literal? Consider to remove it.